packages/python/plotly/plotly/express/_core.py

```diff
@@ -304,26 +291,28 @@ def make_trace_kwargs(args, trace_spec, g, mapping_labels, sizeref):
                     result["z"] = g[v]
                     result["coloraxis"] = "coloraxis1"
                     mapping_labels[v_label] = "%{z}"
-                elif trace_spec.constructor in [go.Sunburst, go.Treemap]:
-                    colorable = "marker"
-                    if colorable not in result:
-                        result[colorable] = dict()
-                    # Other if/else block
+                elif trace_spec.constructor in [
+                    go.Sunburst,
+                    go.Treemap,
+                    go.Pie,
+                    go.Funnelarea,
+                ]:
+                    if "marker" not in result:
+                        result["marker"] = dict()
+
                     if args.get("color_is_continuous"):
-                        result[colorable]["colors"] = g[v]
-                        result[colorable]["coloraxis"] = "coloraxis1"
+                        result["marker"]["colors"] = g[v]
+                        result["marker"]["coloraxis"] = "coloraxis1"
                         mapping_labels[v_label] = "%{color}"
                     else:
-                        result[colorable]["colors"] = make_color_mapping(
-                            g[v], args["color_discrete_sequence"]
-                        )
-                elif trace_spec.constructor in [go.Pie, go.Funnelarea]:
-                    colorable = "marker"
-                    if colorable not in result:
-                        result[colorable] = dict()
-                    result[colorable]["colors"] = make_color_mapping(
-                        g[v], args["color_discrete_sequence"]
-                    )
+                        result["marker"]["colors"] = []
+                        mapping = {}
+                        for cat in g[v]:
+                            if mapping.get(cat) is None:
+                                mapping[cat] = args["color_discrete_sequence"][
+                                    len(mapping) % len(args["color_discrete_sequence"])
+                                ]
+                            result["marker"]["colors"].append(mapping[cat])
```

Comment:
Oh interesting, I did not know that `marker_colors` could be a dictionary, I thought it had to be a list/array.

Reply:
I think it does have to be a list, and in this case it is...?

Reply:
this is the same code you had, just inlined :)

Reply:
oh yes I had read too fast!